### PR TITLE
Change some dynamics to Object? and document slot better

### PR DESCRIPTION
**Pull request by Hixie**
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/framework.dart [hunk 1]
```diff
@@ -3038,10 +3038,11 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   /// Information set by parent to define where this child fits in its parent's
   /// child list.
   ///
-  /// Subclasses of Element that only have one child should use null for
-  /// the slot for that child.
```
Comment (xu-baolin) [inline packages/flutter/lib/src/widgets/framework.dart:3042]: Maybe we should remain this info or remove the null check such as `SingleChildRenderObjectElement.insertRenderObjectChild`.

Comment (Hixie) [reply packages/flutter/lib/src/widgets/framework.dart:3042]: as far as i can tell this paragraph was just wrong. most widgets with one child should pass through their own slot to their child, and for the single-child render object widgets it doesn't really matter in most cases but there can definitely be cases where null is the wrong answer. I figured the existing docs in RenderObjectElement were a better source of information.

Comment (xu-baolin) [reply packages/flutter/lib/src/widgets/framework.dart:3042]: Make sense.

**Review by xu-baolin — APPROVED** [review]
LGTM

**Comment by Hixie** [comment]
Thanks for the review! Will land on green if the tests do indeed go green (they might not...).

**Pull request merged**